src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -2258,13 +2258,8 @@ protected override void NewItem(
                         {
                             exists = true;
 
-                            var normalizedTargetPath = strTargetPath;
-                            if (strTargetPath.StartsWith(".\\", StringComparison.OrdinalIgnoreCase) ||
-                                strTargetPath.StartsWith("./", StringComparison.OrdinalIgnoreCase))
-                            {
-                                normalizedTargetPath = Path.Join(SessionState.Internal.CurrentLocation.ProviderPath, strTargetPath.AsSpan(2));
-                            }
-
+                            // check if the target is a file or directory
+                            var normalizedTargetPath = Path.Combine(Path.GetDirectoryName(path), strTargetPath);
```

Comment:
When the user specified target path is like `./dir` or `.\dir`, the value returned from `Path.Combine` would be like `<sym-dir-path>{directory-separator}./dir` or `<sym-dir-path>{directory-separator}.\dir`. This may not be a problem on Windows (which I'm not certain), but on Unix platform, a path like `/home/user-a/.\temp` cannot be resolved to `/home/user-a/temp`. 

![image](https://github.com/user-attachments/assets/3c71a031-bf4d-45d3-8550-f4e2dd3235ae)

So, I think we should still remove the `.\` or `./` prefix for a relative path.

Also, do we want to normalize directory separators used in the `-Target`? I mean, if user specifies `.\subdir\dir` on Linux, shall we normalize that to be `./subdir/dir`?

Reply:
> Also, do we want to normalize directory separators used in the `-Target`? I mean, if user specifies `.\subdir\dir` on Linux, shall we normalize that to be `./subdir/dir`?

There are already issues with path normalization on Unix since `\` is valid char there. I think we shouldn't add one more.

Reply:
Thanks for the review, these are both good observations, although the solution to the issues is a bit out of scope of the original goal of this PR (but I'm open to implementing them as part of this PR).

Just writing down my thoughts for the implementation:
- The path to the symlink is passed already resolved and pre-processed, but the `-Target`/`-Value` is passed as-is, since the previous layers don't know how it's used.
- PowerShell supports both \ and / as directory separators on all platforms. Therefore, we should convert separators for the value that we store in the symlink target to match what rest of the shell does, although it prevents the user from symlinking to paths that contain `\` on Linux/Mac.
- Personally, I wouldn't do any other processing on `-Target`. Don't see any reason to remove `./`/`.\`, since they should work fine as long as the separators are correct.
- Since the file/directory distinction for symlinks is not a thing on Linux/Mac, I'd personally just remove the whole logic around looking up the target and only use it on Windows.

What needs to be done:
- unify slashes in `-Target` with the correct platform-specific separator
- wrap target type check in `#if` for Windows only

@daxian-dbw Do you agree?

Reply:
@God-damnit-all In that case, just don't pass paths with `./` as `-Target`. Other people may have the opposite preference, and since both options behave identically in practice, I don't see any reason to force one or the other.